encord/client.py

```diff
@@ -454,18 +503,58 @@ def upload_video(
         """
         This function is documented in :meth:`encord.dataset.Dataset.upload_video`.
         """
-        if os.path.exists(file_path):
-            signed_urls = upload_to_signed_url_list(
-                [file_path], self._config, self._querier, Video, cloud_upload_settings=cloud_upload_settings
+        if not os.path.exists(file_path):
+            raise encord.exceptions.EncordException(message=f"{file_path} does not point to a file.")
+
+        signed_url = upload_to_signed_url_list(
+            [file_path],
+            self._config,
+            self._querier,
+            Video,
+            cloud_upload_settings=cloud_upload_settings,
+        )[0]
+
+        upload_job_id = self._querier.basic_setter(
+            DatasetDataLongPolling,
+            uid=self._querier.resource_id,
+            payload=DatasetDataLongPollingParams(
+                data_items=DataUploadItems(
+                    videos=[
+                        DataUploadVideo(
+                            object_url=signed_url["file_link"],
+                            title=title or signed_url["title"],
+                        )
+                    ],
+                ),
+                files=None,
+                integration_id=None,
+                ignore_errors=False,
+                folder_uuid=folder_uuid,
+                file_name=None,
+            ),
+        )["process_hash"]
+
+        res = self.__add_data_to_dataset_get_result(upload_job_id)
+
+        if res.status == LongPollingStatus.DONE:
+            if len(res.data_hashes_with_titles) != 1:
+                encord.exceptions.EncordException(
+                    f"An error has occurred during video upload. len({res.data_hashes_with_titles=}) != 1"
+                )
+
+            res_item = res.data_hashes_with_titles[0]
+            logger.info("Upload complete.")
+
+            return Video(  # Video model types annotations are not correct
+                {
+                    "data_hash": res_item.data_hash,
+                    "title": res_item.title,
+                }
             )
-            res = upload_video_to_encord(signed_urls[0], title, folder_uuid, self._querier)
-            if res:
-                logger.info("Upload complete.")
-                return Video(res)
-            else:
-                raise encord.exceptions.EncordException(message="An error has occurred during video upload.")
+        elif res.status == LongPollingStatus.ERROR:
+            raise encord.exceptions.EncordException(f"An error has occurred during video upload. {res.errors}")
         else:
-            raise encord.exceptions.EncordException(message=f"{file_path} does not point to a file.")
+            raise ValueError(f"res.status={res.status}, this should never happen")
```

Comment:
More like `RuntimeError`; the `ValueError` is "you called us with the wrong stuff", and here it's very much our own fault, no the user's ;)

Also probably don't say "this should never happen". It's true, but it's not great customer communication!



```suggestion
            raise RuntimeError(f"Job in unexpected res.status={res.status} after the timeout")
```